angular/projects/admin-nrpti/src/app/mines/mines-collections-list-resolver.ts

```diff
@@ -44,6 +44,36 @@ export class MinesCollectionsListResolver implements Resolve<Observable<object>>
       and['project'] = params.mineId;
     }
 
+    if (params.type) {
+      or['type'] = params.type.join(',');
+    }
+
+    if (params.agency) {
+      or['agency'] = params.agency.join(',');
+    }
+
+    if (params.bcmiTabType) {
+      if (!or['type']) {
```

Comment:
With the check on line 72, does this need to be here?

Reply:
Yeah, looks weird but is required. Basically I create a 'stub' or object and populate if needed. It's possible it might be empty by the time the checks are done, so in that case I just make sure to delete the type ref so it isn't going to inadvertently search for null types and return incorrect results